### ISSUE TITLE
My saved data augmentation got worse result

### ISSUE THREAD

**Issue by datvuthanh** [ultralytics/yolov5#5477]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi guys,

I have a problem about data augmentation. Firstly, I used this code dataset.py to save all image augmentation files to folder (jpg). Next, I trained model with this augmentation dataset and set up augment=False. This problem is, if I setup augment=True and directly train dataset augmentation without saving, I got the better results than dataset augmentation which I saved. I don't know why results are different, whereas I checked results between two approaches many times.

Thank you.

**Comment by glenn-jocher** [comment]
@datvuthanh 👋 Hello! Thanks for asking about **image augmentation**. YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the testloader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch.yaml
```
https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/data/hyp.scratch.yaml#L1-L33

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
> <img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="700">

Good luck and let us know if you have any other questions!

**Comment by datvuthanh** [comment]
I understand what you said, for example, you showed me 16 images augmentation. I saved all images augmentation in folder which will replace my dataset. After I train this dataset augmentation (augment=False), and I cannot achieve the results like directly train dataset augmentation without saving (augment=True). Do you understand?

**Comment by glenn-jocher** [comment]
@datvuthanh yes I understand. You've tried an experiment which returned a worse result, therefore the conclusion of your experiment is that the default training script works best. Go with the default training script.

**Comment by datvuthanh** [comment]
@glenn-jocher Thank glenn, but I want to know why the results are worse. My autoanchor without saving is 5.33 (original size), but my autoanchor with saved dataset (640x640) is only 4.3. Why? I see you calculate anchorboxes with original size (not 640x640 or 1280x1280) right?

**Comment by glenn-jocher** [comment]
@datvuthanh I'm sorry, we generally don't comment on users custom training results.
